src/transaction/utils.ts

```diff
@@ -412,7 +417,130 @@ function convertMemo(memo: Memo): { Memo: ApiMemo } {
   };
 }
 
+/**
+ * Autofills fields in a transaction.
+ *
+ * @param client - A client.
+ * @param tx - A transaction to autofill fields.
+ * @returns An autofilled transaction.
+ */
+async function autofill(client: Client, tx: Transaction): Promise<Transaction> {
```

Comment:
This file should go in the `ledger` folder - something like `src/ledger/autofill.ts`
(yes, this doesn't really have to do anything with the ledger, but that's the folder all the sugar is going in, and it'll be renamed soon)